crates/nu-cli/src/cli.rs

```diff
@@ -468,6 +470,27 @@ pub fn register_plugins(context: &EvaluationContext) -> Result<(), ShellError> {
     Ok(())
 }
 
+fn reload_cfg_if_modified_and_print(ctx: &EvaluationContext) {
+    if let Err(err) = reload_config_if_modified(&ctx) {
+        ctx.host.lock().print_err(
+            ShellError::untagged_runtime_error(format!(
+                "Could not reload your config.\nThe error was: {:?}",
+                err
+            )),
+            &Text::from("".to_string()),
+        );
+    }
+}
+
+fn reload_config_if_modified(ctx: &EvaluationContext) -> Result<(), ShellError> {
+    if let Some(global_cfg) = &mut ctx.configs.lock().global_config {
+        if global_cfg.is_modified()? {
```

Comment:
We only reload the config if it is modified. (Otherwise the old values are still up to date)